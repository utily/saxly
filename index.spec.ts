import "jest"
import { saxly } from "./index"

describe("saxly", () => {
	it("parse <root/>", async () => expect(await saxly.parse("<root/>")).toMatchSnapshot())
	it("parse <root></root>", async () => expect(await saxly.parse("<root></root>")).toMatchSnapshot())
	it('parse <root attribute="value"></root>', async () =>
		expect(await saxly.parse('<root attribute="value"></root>')).toMatchSnapshot())
	it("parse <root readonly></root>", async () => expect(await saxly.parse("<root readonly></root>")).toMatchSnapshot())
	it("parse <root>text</root>", async () => expect(await saxly.parse("<root>text</root>")).toMatchSnapshot())
	it("parse <root>first name</root>", async () =>
		expect(await saxly.parse("<root>first name</root>")).toMatchSnapshot())
	it('parse <?xml version="1.0" language="en/US"?>\n<root/>', async () =>
		expect(await saxly.parse('<?xml version="1.0" language="en/US"?>\n<root/>')).toMatchSnapshot())
	it("parse <root><nested/></root>", async () => expect(await saxly.parse("<root><nested/></root>")).toMatchSnapshot())
	it("parse <root><nested></nested></root>", async () =>
		expect(await saxly.parse("<root><nested></nested></root>")).toMatchSnapshot())
	it("parse <root>first<nested></nested>last</root>", async () =>
		expect(await saxly.parse("<root>first<nested></nested>last</root>")).toMatchSnapshot())
	it("parse <root><nested><inner/></nested></root>", async () =>
		expect(await saxly.parse("<root><nested><inner/></nested></root>")).toMatchSnapshot())
	it("parse <root>first<nested>middle</nested>last</root>", async () =>
		expect(await saxly.parse("<root>first<nested>middle</nested>last</root>")).toMatchSnapshot())
	it('parse <root attribute="value">\nText<element key="value" readonly>last name</element></root>', async () =>
		expect(
			await saxly.parse('<root attribute="value">\nText<element key="value" readonly>last name</element></root>')
		).toMatchSnapshot())
	it("parse <root>  first middle last</root>", async () =>
		expect(await saxly.parse("<root>\n  first middle last\n</root>")).toMatchSnapshot())
})
