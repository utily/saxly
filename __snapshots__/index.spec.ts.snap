// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saxly parse <?xml version="1.0" language="en/US"?>
<root/> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root attribute="value">
Text<element key="value" readonly>last name</element></root> 1`] = `
{
  "root": {
    "attributes": {
      "attribute": "value",
    },
    "content": [
      "
",
      "Text",
      {
        "attributes": {
          "key": "value",
          "readonly": undefined,
        },
        "content": [
          "last",
          " ",
          "name",
        ],
        "errors": [],
        "name": "element",
      },
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root attribute="value"></root> 1`] = `
{
  "root": {
    "attributes": {
      "attribute": "value",
    },
    "content": [],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root readonly></root> 1`] = `
{
  "root": {
    "attributes": {
      "readonly": undefined,
    },
    "content": [],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root/> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root>  first middle last</root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      "
  ",
      "first",
      " ",
      "middle",
      " ",
      "last",
      "
",
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root></root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root><nested/></root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      {
        "attributes": {},
        "content": [],
        "errors": [],
        "name": "nested",
      },
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root><nested></nested></root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      {
        "attributes": {},
        "content": [],
        "errors": [],
        "name": "nested",
      },
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root><nested><inner/></nested></root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      {
        "attributes": {},
        "content": [
          {
            "attributes": {},
            "content": [],
            "errors": [],
            "name": "inner",
          },
        ],
        "errors": [],
        "name": "nested",
      },
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root>first name</root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      "first",
      " ",
      "name",
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root>first<nested></nested>last</root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      "first",
      {
        "attributes": {},
        "content": [],
        "errors": [],
        "name": "nested",
      },
      "last",
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root>first<nested>middle</nested>last</root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      "first",
      {
        "attributes": {},
        "content": [
          "middle",
        ],
        "errors": [],
        "name": "nested",
      },
      "last",
    ],
    "errors": [],
    "name": "root",
  },
}
`;

exports[`saxly parse <root>text</root> 1`] = `
{
  "root": {
    "attributes": {},
    "content": [
      "text",
    ],
    "errors": [],
    "name": "root",
  },
}
`;
